common/src/main/java/com/wynntils/core/chat/tabs/ChatTab.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.chat.tabs;
+
+import com.wynntils.core.chat.RecipientType;
+import com.wynntils.mc.event.LocalMessageEvent;
+import com.wynntils.wynn.event.ChatMessageReceivedEvent;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class ChatTab {
+    private String name;
+    private boolean lowPriority;
+
+    // Filters
+    private Set<RecipientType> filteredTypes;
+    private String customRegexString;
+
+    private transient Pattern customRegex;
+
+    public ChatTab(String name, boolean lowPriority, Set<RecipientType> filteredTypes, String customRegexString) {
+        this.name = name;
+        this.lowPriority = lowPriority;
+        this.filteredTypes = filteredTypes;
+        this.customRegexString = customRegexString;
+    }
+
+    public boolean matchMessageFromEvent(ChatMessageReceivedEvent event) {
+        if (filteredTypes != null && !filteredTypes.isEmpty() && !filteredTypes.contains(event.getRecipientType())) {
+            return false;
+        }
+
+        return customRegexString == null
+                || getCustomRegex().matcher(event.getCodedMessage()).matches();
+    }
+
+    public boolean matchMessageFromEvent(LocalMessageEvent event) {
+        if (customRegexString == null) {
+            return false;
```

Comment:
This should be true..?

Reply:
If we don't have a custom regex, we are not matching this. Local messages are command responses (and every mod to player chat message) usually, so they are not matched to RecipientTypes. So if we don't have a regex for a chat tab, we are not getting any command responses there.

Reply:
Hm.

Maybe we should treat these messages as a special recipient category, COMMANDS?

Reply:
> I think a more "flow oriented" filtering would be more readable. Like:
> 
> * "if we have a recipient type filter, and it does not match, return false"
> * "if we have a regexp filter, and it does not match, return false"
> * "otherwise, return true"
> 
> And for local messages, which do not have a recipient, will always match true for the first case.

This doesnt capture the case correctly when we have a regex filter but do not match it.

Reply:
Oh, maybe it does. Im tired

Reply:
It won't work for local messages, because it wont handle no regex case.

Reply:
I'm a bit tired as well, so maybe I'm not thinking straight either. :) But my understanding is that we, confusingly, have two different meanings of "local" right now:

1) chat messages sent from Wynncraft, written by players nearby

2) chat messages sent client-side only, from e.g. mods and commands

This is confusing in itself. I suggest we rename one of them. Either the Wynncraft concept to "NEARBY", or the client-side one to "COMMAND" or "CLIENTSIDE" or something like that. Or, possibly, rename both and have no "local" if it can be misunderstood.

Reply:
Renamed to ClientsideMessageEvent.

Reply:
And then, for 2 (client-side), I think things should work just the same. If we get e.g. an `Unknown command` back as a clientside reply, and it is tagged `MessageRecipientType.CLIENTSIDE`, then it would go to a "Clientside" tab for that recipient type. And if you set up a custom filter with both CLIENTSIDE and a regexp, it would only match messages that fullfil both requirements. So I think it would work just like the rest of the messages.

(In fact, it should probably even be the responsibility of the ChatManager to catch these messages, and tag them properly)

Reply:
Okay. I will refactor this tomorrow. Better for both of us :)

Reply:
I think I did this. I did not mix client sided messages with chatrecievedevent, I think that is fine. LMK if you want anything else changed.

Reply:
I think it looks good now.